### PR TITLE
ADR on Fabric E2E Sample CICD flow

### PR DESCRIPTION
**Pull request by devlace**
# Type of PR

- Documentation changes

## Purpose

- Document decision on which Fabric CICD option we will build as part of milestone 1
- Added the CICD diagram into the Fabric E2E sample readme

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [X] No PII in logs
- [ ] Made corresponding changes to the documentation

## Validation steps

- ...

## Issues Closed or Referenced

- References #790

### CODE REVIEW

**Review by thesqlpro — APPROVED** [review]
Looks good.
I'm assuming Option 3 and 4 are still to be added. Otherwise the page is a great starting point!

**Comment by promisinganuj** [comment]
As per https://github.com/Azure-Samples/modern-data-warehouse-dataops/blob/main/e2e_samples/fabric_dataops_sample/docs/adr_naming_convention.md#github-files-naming-convention, images should be "kebab-case".

File: e2e_samples/fabric_dataops_sample/README.md [hunk 1]
```diff
@@ -40,6 +40,14 @@ At a high level, the deployment sequence proceeds as follows:
 
 Please follow the [How to use the sample](#how-to-use-the-sample) section for detailed instructions.
 
+### Continuous Integration and Continuous Delivery (CI/CD)
```
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/README.md:43]: This should automatically update TOC

File: e2e_samples/fabric_dataops_sample/README.md [hunk 2]
```diff
@@ -40,6 +40,14 @@ At a high level, the deployment sequence proceeds as follows:
...
+
+Fabric has a number of options for CI/CD workflows as documented [here](https://learn.microsoft.com/fabric/cicd/manage-deployment). This utilizes Option 1 - Git-based deployments.
+
+The following shows the overall CI/CD process end to end.
```
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/README.md:47]:
```suggestion
The diagram below illustrates the complete end-to-end CI/CD process:
```

File: e2e_samples/fabric_dataops_sample/README.md [hunk 3]
```diff
@@ -40,6 +40,14 @@ At a high level, the deployment sequence proceeds as follows:
 
 Please follow the [How to use the sample](#how-to-use-the-sample) section for detailed instructions.
 
+### Continuous Integration and Continuous Delivery (CI/CD)
+
+Fabric has a number of options for CI/CD workflows as documented [here](https://learn.microsoft.com/fabric/cicd/manage-deployment). This utilizes Option 1 - Git-based deployments.
```
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/README.md:45]:
```suggestion
Microsoft Fabric has a number of CI/CD workflow options as documented [here](https://learn.microsoft.com/fabric/cicd/manage-deployment). This sample utilizes [Option 1: Git-based deployment](https://learn.microsoft.com/fabric/cicd/manage-deployment#option-1---git--based-deployments).
```

File: e2e_samples/fabric_dataops_sample/README.md (hunk 1 above)
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/README.md:43]: Shall we move it after "Deployed Resources"? Somehow the ordering of headers feels off.

File: e2e_samples/fabric_dataops_sample/docs/adr_fabric_cicd.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+|---------------------------|----------|----------|----------|----------|
+| Technical simplicity      | Medium   | Low      | High     | Low      |
+| Viability of E2E solution | High     | High     | Low      | Medium   |
+| Functionality offered     | High     | High     | Medium   | High     |
+| Long-term relevance       | High     | Medium   | High     | Medium   |
+| Existing samples          | No       | No       | Yes      | No       |
+
+## Proposal details
+
+Based on the key consideration, **option 1** and **option 2** were considered extensively. However, due to the very high technical complexity of option 2 specifically around item dependency management and need to track individual item IDs manually, Option 1 is the preferred **Option 1** for this milestone.
```
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/docs/adr_fabric_cicd.md:31]:
```suggestion
Based on the key considerations, both **Option 1** and **Option 2** were extensively analyzed. However, due to the significant technical complexity of Option 2, particularly in managing item dependencies and manually tracking individual item IDs, Option 1 has been selected as the preferred approach for this milestone.
```

File: e2e_samples/fabric_dataops_sample/README.md (hunk 1 above)
Comment (devlace) [reply e2e_samples/fabric_dataops_sample/README.md]: I've rearranged a few things. I think this flow is a bit better and a bit more aligned with our existing databricks sample readme.

**Review by promisinganuj — APPROVED** [review]
LGTM!

We might have to review the end-of-file hook for ".drawio" files later.

File: e2e_samples/fabric_dataops_sample/images/fabric-cicd-option2.drawio
Comment (promisinganuj) [inline e2e_samples/fabric_dataops_sample/images/fabric-cicd-option2.drawio:1]: @devlace , this file is empty.

**Review by promisinganuj — APPROVED** [review]
lgtm!

**Pull request merged**